cmd/peer-s3-client.go

```diff
@@ -185,6 +185,14 @@ func (sys *S3PeerSys) ListBuckets(ctx context.Context, opts BucketOptions) ([]Bu
 				}
 			}
 		}
+		// loop through buckets and see if some with lost quorum
+		// these could be stale buckets lying around, remove them
+		for bktName, count := range bucketsMap {
+			if count < quorum {
+				// Its safe to remove as its stale bucket lying around
+				sys.DeleteBucket(ctx, bktName, DeleteBucketOptions{Force: true})
```

Comment:
quorum issue can happen where nodes are down, we cannot just assume that to be a precursor to delete a bucket entirely.

we should carefully look at the deleteDangling code for objects. We must be absolutely sure of what we are trying to do here.

This PR makes quorum less of a definitive way to purge the entire data automatically which would be unexpected.

Reply:
sure, makes sense. let me explore more and redo this.

Reply:
I think let's try to avoid calling Deletebucket with Force set to true; It is okay if it fails to delete a bucket which is stale rather than always doing a force delete.

isObjectDangling(), this is a function that decides if an object is dangling or not; I think the logic is we only mark a bucket as dangling in well defined use cases; meaning if we do not have certain errors and secnarios, we should not dare to remove a bucket.